addons/purchase/views/account_move_views.xml

```diff
@@ -14,7 +14,7 @@
                 <field name="purchase_vendor_bill_id" nolabel="1"
                        attrs="{'invisible': ['|', ('state','!=','draft'), ('move_type', '!=', 'in_invoice')]}"
                        class="oe_edit_only"
-                       domain="partner_id and [('company_id', '=', company_id), ('partner_id','child_of', [partner_id])] or [('company_id', '=', company_id)]"
+                       domain="partner_id and [('company_id', '=', company_id), ('partner_id.commercial_partner_id', '=', commercial_partner_id)] or [('company_id', '=', company_id)]"
```

Comment:
```suggestion
                       domain="partner_id and [('company_id', '=', company_id), ('partner_id', 'child_of', commercial_partner_id)] or [('company_id', '=', company_id)]"
```
This could cover more cases

Reply:
@yhu-odoo I'm quite sure that covers the same cases.
From the performance PoV I wonder if the usage of "child_of" is more costly to the database query.

Reply:
Ok, you're right. The only difference is when parent and child are both companies, but I think in that case it shouldn't be shown.